src/box/lua/schema.lua

```diff
@@ -447,14 +464,16 @@ box.atomic = function(arg0, arg1, ...)
         if not utils.is_callable(arg1) then
             box.error(box.error.ILLEGAL_PARAMS, usage, 2)
         end
-        call_at(2, box.begin, arg0)
-        return atomic_tail(1, pcall(arg1, ...))
+        local options = arg0
+        check_param_table(options, atomic_options, 2)
+        call_at(2, box_begin_impl, options)
```

Comment:
I didn't get, why don't we pass `level ` to `box_begin_impl` here

Reply:
`call_at` takes level as the first argument. We can also pass it to `begin_impl`, but `call_at` overrides the level anyway.

Reply:
Actually, thanks for the comment - I noticed that I don't need `call_at` here. I am passing the level directly to `box_begin_impl` now.